### PR TITLE
Scala 2.13.0 Cats-Effect 2.0.0-RC1

### PR DESCRIPTION
**Pull request by Avasil**
Related to #862

@alexandru @oleg-py We have to figure out whether our next release is on top of `cats-effect 2.0.0-M1` and/or `1.3.0`. I think there are some binary incompatibilities in `cats-effect` 2.0

### CODE REVIEW

**Comment by oleg-py** [comment]
> I think there are some binary incompatibilities in cats-effect 2.0
IIRC cats-effect 2.0 is binary compatible with 1.x on Scala 2.12, same as cats.

3.0 final will be on cats-effect 2.0, but we can roll RC3 with fixed locals for CE 1.x if CE 2.0 isn't yet final by that time.

**Comment by xuwei-k** [comment]
Any updates? 👀

**Comment by Avasil** [comment]
@xuwei-k I don't know why CI fails but we will probably wait with release for stable Cats-Effect support if we use 2.0 for Monix 3.0 (or 3.1)

Would it be helpful to release a snapshot version from this branch?

**Comment by niebloomj** [comment]
Any updates?

**Comment by Avasil** [comment]
@niebloomj I have a snapshot from this branch if you're interested: `3.0.0-eddc79d-SNAPSHOT`

I'm not sure when stable `Cats-Effect` supporting `2.13` is released and I don't want to release on a milestone

File: project/MimaFilters.scala [hunk 1]
```diff
@@ -2,6 +2,16 @@ import com.typesafe.tools.mima.core.ProblemFilters.exclude
 import com.typesafe.tools.mima.core._
 
 object MimaFilters {
+  lazy val changesFor_3_0_0__RC5: Seq[ProblemFilter] = Seq(
+    // Incompatible signatures should not cause linking problems.
```
Comment (Avasil) [inline project/MimaFilters.scala]: Should be fine but we might want to repeat this test: https://github.com/typelevel/cats-effect/pull/609#pullrequestreview-280142297

**Comment by alexandru** [comment]
@Avasil see my PR: https://github.com/Avasil/monix/pull/4

**Comment by alexandru** [comment]
The tests are now failing due to Mima, since we never published packages for 2.13.0.

**Comment by alexandru** [comment]
We should only run Mima for the lowest version supported, in this case 2.11 and not for 2.13.

File: project/MimaFilters.scala (hunk 1 above)
Comment (alexandru) [reply project/MimaFilters.scala]: Are we sure they don't cause linking problems?

What happened there?

Comment (alexandru) [reply project/MimaFilters.scala]: Oh, it's due to the changed `Parallel` from Cats or something?

Comment (alexandru) [reply project/MimaFilters.scala]: If so, then there's nothing we can do there.

**Pull request merged**